src/bindings/python/src/openvino/frontend/tensorflow/graph_iterator.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (C) 2018-2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+# flake8: noqa
+# mypy: ignore-errors
+
+import tensorflow as tf
+from openvino.frontend.tensorflow.py_tensorflow_frontend import _FrontEndPyGraphIterator as GraphIterator
+
+from .node_decoder import TFGraphNodeDecoder
+
+
+class GraphIteratorTFGraph(GraphIterator):
+    def __init__(self, tf_graph, inner_graph=False):
+        GraphIterator.__init__(self)
+        self.m_graph = tf_graph
+        self.m_node_index = 0
+        self.m_decoders = []
+        self.m_inner_graph = inner_graph
+
+        for op in tf_graph.get_operations():
+            self.m_decoders.append(TFGraphNodeDecoder(op, inner_graph))
+
+        self.m_iterators = {}
+        for func_name, _ in self.m_graph._functions.items():
+            self.m_iterators[func_name] = None
+
+    def get_input_names(self) -> list:
+        inp_ops = filter(lambda op: op.type == "Placeholder" and len(op.inputs) == 0, self.m_graph.get_operations())
+        inp_names = []
+        for inp in inp_ops:
+            type_attr = inp.node_def.attr["dtype"].type
+            if tf.dtypes.DType(type_attr).name != "resource" or self.m_inner_graph:
```

Comment:
Not sure if we should exclude all such Placeholders in the main graph. Can you leave a comment here what exactly we want to filter out here?

Reply:
Added comment here.